### PR TITLE
Multiclass support in oneAPI interfaces for SVM

### PR DESCRIPTION
**Pull request by PetrovKP**
1. Enable multiclass support for SVM in oneDAL part
2. Merge to SVM model from Multiclass model in oneDAL part
2. Fix SVM Regression issues with accuracy
3. Deprecated `bias` for model and result and introduce `biases` instead one
4. Add multiclass example for host part
2. Delete v1 interfaces for Multiclass and SVM in DAAL interfaces

### CODE REVIEW

File: cpp/oneapi/dal/algo/svm/backend/cpu/train_cls_kernel.cpp [hunk 1]
```diff
@@ -41,12 +42,19 @@ using descriptor_t = detail::descriptor_base<task::classification>;
...
 template <typename Float, daal::CpuType Cpu, typename Method>
 using daal_svm_kernel_t =
     daal_svm::training::internal::SVMTrainImpl<to_daal_method<Method>::value, Float, Cpu>;
 
+template <typename Float, daal::CpuType Cpu>
+using daal_multiclass_kernel_t = daal_multiclass::training::internal::
+    MultiClassClassifierTrainKernel<daal_multiclass::training::oneAgainstOne, Float, Cpu>;
+
 template <typename Float, typename Method>
 static result_t call_daal_kernel(const context_cpu& ctx,
```
Comment (rlnx) [inline cpp/oneapi/dal/algo/svm/backend/cpu/train_cls_kernel.cpp:59]: Suggest to think how to split this function, now it's ~120 lines of code.

File: cpp/oneapi/dal/algo/svm/common.hpp [hunk 2]
```diff
@@ -331,15 +331,21 @@ class model : public base {
...
-    double get_bias() const;
+    [[deprecated("Use get_biases() instead.")]] double get_bias() const;
 
-    auto& set_bias(double value) {
+    [[deprecated("Use set_biases() instead.")]] auto& set_bias(double value) {
         set_bias_impl(value);
         return *this;
     }
 
+    /// A $class_count*(class_count-1)/2 \\times 1$ table constants in decision function
```
Comment (rlnx) [inline cpp/oneapi/dal/algo/svm/common.hpp:341]:
```suggestion
    /// A $k * (k - 1) / 2 \\times 1$ table constants in decision function, where $k$ denotes the number of classes.
```

Comment (rlnx) [reply cpp/oneapi/dal/algo/svm/common.hpp:341]: `class_count`. What about regression problem?

File: cpp/oneapi/dal/algo/svm/train_types.cpp [hunk 3]
```diff
@@ -102,7 +102,13 @@ const table& train_result<Task>::get_coeffs() const {
 
 template <typename Task>
 double train_result<Task>::get_bias() const {
-    return impl_->trained_model.get_bias();
+    // return dal::detail::get_impl(impl_->trained_model).bias;
+    return 0.0;
```
Comment (rlnx) [inline cpp/oneapi/dal/algo/svm/train_types.cpp:106]: Implementation should be valid despite this function is deprecated.

**Comment by PetrovKP** [comment]
/intelci: run
set run_test_dal4py=1

**Comment by PetrovKP** [comment]
/intelci: run
set run_test_dal4py=1

**Comment by PetrovKP** [comment]
/intelci: run
set run_test_dal4py=1
set compatibility_branch=dev/2020-svm-fix

File: cpp/oneapi/dal/algo/svm/backend/cpu/train_cls_kernel.cpp (hunk 1 above)
Comment (PetrovKP) [reply cpp/oneapi/dal/algo/svm/backend/cpu/train_cls_kernel.cpp]: done

File: cpp/oneapi/dal/algo/svm/train_types.cpp (hunk 3 above)
Comment (PetrovKP) [reply cpp/oneapi/dal/algo/svm/train_types.cpp]: done

**Comment by PetrovKP** [comment]
/intelci: run
set run_test_dal4py=1
set compatibility_branch=dev/2020-svm-fix
set infra_branch=dev/multiclass-v1-del

File: cpp/oneapi/dal/algo/svm/common.hpp (hunk 2 above)
Comment (PetrovKP) [reply cpp/oneapi/dal/algo/svm/common.hpp]: Apply

File: cpp/daal/src/algorithms/svm/svm_train_thunder_impl.i [hunk 4]
```diff
@@ -415,6 +402,7 @@ services::Status SVMTrainImpl<thunder, algorithmFPType, cpu>::SMOBlockSolver(con
             gradLocal[i] += delta * (KiBi - KiBj);
         }
     }
+    // printf("[LOCAL] iter: %lu localDiff: %.4lf\n", iter, localDiff);
```
Comment (owerbat) [inline cpp/daal/src/algorithms/svm/svm_train_thunder_impl.i:405]:
```suggestion
```

File: cpp/daal/src/algorithms/multiclassclassifier/multiclassclassifier_train_oneagainstone_impl.i [hunk 5]
```diff
@@ -124,17 +156,126 @@ services::Status MultiClassClassifierTrainKernel<oneAgainstOne, algorithmFPType,
...
+            for (size_t j = 0; j < nVectors; ++j)
+            {
+                const size_t label = size_t(y[j]);
+                if (isSVData[j] && label == iClass)
+                {
+                    ++sv_counts[iClass];
+                }
+            }
+            nSV += sv_counts[iClass];
+            // printf("%lu: %lu\n", iClass, sv_counts[iClass]);
```
Comment (owerbat) [inline cpp/daal/src/algorithms/multiclassclassifier/multiclassclassifier_train_oneagainstone_impl.i:198]:
```suggestion
```

File: cpp/daal/src/algorithms/multiclassclassifier/multiclassclassifier_train_oneagainstone_impl.i [hunk 6]
```diff
@@ -124,17 +156,126 @@ services::Status MultiClassClassifierTrainKernel<oneAgainstOne, algorithmFPType,
...
+        }
+        printf("sumSV: %lu\n", nSV);
```
Comment (owerbat) [inline cpp/daal/src/algorithms/multiclassclassifier/multiclassclassifier_train_oneagainstone_impl.i:200]:
```suggestion
```

File: cpp/daal/src/algorithms/svm/svm_train_thunder_impl.i [hunk 7]
```diff
@@ -164,21 +162,11 @@ services::Status SVMTrainImpl<thunder, algorithmFPType, cpu>::compute(const Nume
...
-    {
-        DAAL_CHECK(alphaTArray.size() >= 2 * nVectors, services::ErrorMemoryAllocationFailed)
-        PRAGMA_IVDEP
-        PRAGMA_VECTOR_ALWAYS
-        for (size_t i = 0; i < nVectors; ++i)
-        {
-            alpha[i] = alpha[i] - alpha[i + nVectors];
-        }
-    }
+    // printf("[global] iter %lu\n", iter);
```
Comment (owerbat) [inline cpp/daal/src/algorithms/svm/svm_train_thunder_impl.i:165]:
```suggestion
```

**Comment by PetrovKP** [comment]
/intelci: run
set run_test_dal4py=1
set compatibility_branch=dev/2020-svm-fix
set infra_branch=dev/multiclass-v1-del

**Review by owerbat — APPROVED** [review]

**Pull request merged**